tests/platformtests/test_base.py

```diff
@@ -0,0 +1,89 @@
+########
```

Comment:
I wouldn't worry about generalising this too much more- we've got a new plugin tests framework we'll be shoving on the plugins in the hopefully near future.